robolectric-resources/src/main/java/org/robolectric/res/PackageResourceTable.java

```diff
@@ -63,24 +80,53 @@ public InputStream getRawValue(int resId, String qualifiers) {
     return getRawValue(getResName(resId), qualifiers);
   }
 
-  public int getPackageIdentifier() {
-    return resourceIndex.getPackageIdentifier();
+  int getPackageIdentifier() {
+    return packageIdentifier;
   }
 
   @Override
-  public void addResource(int resId, String type, String name) {
-    resourceIndex.addResource(resId, type, name);
+  public void receive(Visitor visitor) {
+    resources.receive(visitor);
   }
 
-  @Override
-  public void receive(Visitor visitor) {
-    data.receive(visitor);
+  // TODO: Merge this method with the addResource(String, String, TypedResource) so that the ID is provided by the
+  // caller. When we start to read arsc files the ID will already be provided so we want to write the id, name, type and
+  // value information in the same call rather than split up like this.
+  synchronized void addResource(int id, String type, String name) {
```

Comment:
Do the `get*()` methods ever change state here? If not, can we remove `synchronized` since everything will be immutable by the time we've read everything in? (and maybe explicitly make the containers immutable?)